hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMSecurityProtocolServer.java

```diff
@@ -225,9 +214,9 @@ public String getSCMCertificate(ScmNodeDetailsProto scmNodeDetails,
   private String getEncodedCertToString(String certSignReq, NodeType nodeType)
       throws IOException {
     Future<CertPath> future;
-    if (nodeType == NodeType.SCM) {
+    if (nodeType == NodeType.SCM && rootCertificateServer != null) {
```

Comment:
With the primordial decoupling, as I understand it, only when the cluster starts is the `rootCA` used to sign the leader `subCA` certificates.  After that it is not used.  Is that handled here, once the subca certs for the scm is signed is the `rootCertificateServer` here `null`?

Reply:
@neils-dev , the basic idea is remove the limitation that only rootCA can sign other SCM sub CA certificate.  The new proposal is SCM leader will sign other SCMs. The rootCertificateServer null check here is a kind of optimization. If the SCM leader also has the rooCA certificate, then use the rootCA certificate to sign the cert request, otherwise, use leader SCM's sub-CA to sign it.

Reply:
Thanks @ChenSammi for clarifying.  Makes sense - remove rootCA signing limitation and sign with rootCA if present on SCM otherwise sign with sub-CA.